internal/service/advancedclustertpf/resource_compatibility_reuse.go

```diff
@@ -11,6 +12,23 @@ import (
 	"go.mongodb.org/atlas-sdk/v20241113002/admin"
 )
 
+func IsMajorVersionHigherGreaterOrEqual(input *string, version float64) bool {
+	if input == nil || *input == "" {
+		return true
+	}
+	parts := strings.SplitN(*input, ".", 2)
+	if len(parts) == 0 {
+		return false
+	}
+
+	value, err := strconv.ParseFloat(parts[0], 64)
+	if err != nil {
+		return false
```

Comment:
when the func caller uses `! IsMajorVersionHigherGreaterOrEqual` it will get true when parse fails, maybe it's not a problem

Reply:
Updated and using a warning when that is the case